From d9486a4aeb3459dd90c233e1c4d85dfc4eadfb95 Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <Ghennadi.Procopciuc@nxp.com>
Date: Fri, 4 Dec 2020 11:51:28 +0200
Subject: [PATCH 02/78] dts: s32-gen1: Fix SIUL2 ranges

This shrinks the allocated ranges for output pads and IRQ registers.

Issue: ALB-6099
Signed-off-by: Ghennadi Procopciuc <Ghennadi.Procopciuc@nxp.com>
---
 .../dts/freescale/fsl-s32g274-simulator.dts   | 25 -------------------
 .../boot/dts/freescale/fsl-s32g274a.dtsi      |  6 ++---
 arch/arm64/boot/dts/freescale/fsl-s32r45.dtsi |  6 ++---
 3 files changed, 6 insertions(+), 31 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-s32g274-simulator.dts b/arch/arm64/boot/dts/freescale/fsl-s32g274-simulator.dts
index e4ac0b7522c6..4016f897b741 100644
--- a/arch/arm64/boot/dts/freescale/fsl-s32g274-simulator.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-s32g274-simulator.dts
@@ -335,31 +335,6 @@
 			status = "okay";
 		};
 	};
-	irq_regs: siul2_reg@0x44010000 {
-		compatible = "fsl,irq_reqs-s32gen1","syscon","simple-mfd";
-		reg = <0x0 0x44010000 0x0 0x44>;
-		little-endian;
-	};
-	opad0_regs: siul2_reg@0x4009D700 {
-		compatible = "fsl,opad0_reqs-s32gen1","syscon","simple-mfd";
-		reg = <0x0 0x4009D700 0x0 0x10>;
-		big-endian;
-	};
-	ipad0_regs: siul2_reg@0x4009D740 {
-		compatible = "fsl,ipad0_reqs-s32gen1","syscon","simple-mfd";
-		reg = <0x0 0x4009D740 0x0 0x10>;
-		big-endian;
-	};
-	opad1_regs: siul2_reg@0x4401170C {
-		compatible = "fsl,opad0_reqs-s32gen1","syscon","simple-mfd";
-		reg = <0x0 0x4401170C 0x0 0x14>;
-		big-endian;
-	};
-	ipad1_regs: siul2_reg@0x4401174C {
-		compatible = "fsl,ipad0_reqs-s32gen1","syscon","simple-mfd";
-		reg = <0x0 0x4401174C 0x0 0x14>;
-		big-endian;
-	};
 };
 
 &cluster0_l2_cache {
diff --git a/arch/arm64/boot/dts/freescale/fsl-s32g274a.dtsi b/arch/arm64/boot/dts/freescale/fsl-s32g274a.dtsi
index 8fe33c23420b..8815191b8581 100644
--- a/arch/arm64/boot/dts/freescale/fsl-s32g274a.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-s32g274a.dtsi
@@ -257,9 +257,9 @@
 			status = "okay";
 		};
 	};
-	irq_regs: siul2_reg@0x44010000 {
+	irq_regs: siul2_reg@0x44010010 {
 		compatible = "fsl,irq_reqs-s32gen1","syscon","simple-mfd";
-		reg = <0x0 0x44010000 0x0 0x44>;
+		reg = <0x0 0x44010010 0x0 0x34>;
 		little-endian;
 	};
 	opad0_regs: siul2_reg@0x4009D700 {
@@ -274,7 +274,7 @@
 	};
 	opad1_regs: siul2_reg@0x4401170C {
 		compatible = "fsl,opad0_reqs-s32gen1","syscon","simple-mfd";
-		reg = <0x0 0x4401170C 0x0 0x14>;
+		reg = <0x0 0x4401170C 0x0 0xC>;
 		big-endian;
 	};
 	ipad1_regs: siul2_reg@0x4401174C {
diff --git a/arch/arm64/boot/dts/freescale/fsl-s32r45.dtsi b/arch/arm64/boot/dts/freescale/fsl-s32r45.dtsi
index fd67f93b1d33..d14dd75dbb32 100644
--- a/arch/arm64/boot/dts/freescale/fsl-s32r45.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-s32r45.dtsi
@@ -206,9 +206,9 @@
 		};
 	};
 
-	irq_regs: siul2_reg@0x4403C000 {
+	irq_regs: siul2_reg@0x4403C010 {
 		compatible = "fsl,irq_reqs-s32gen1","syscon","simple-mfd";
-		reg = <0x0 0x4403C000 0x0 0x40>;
+		reg = <0x0 0x4403C010 0x0 0x30>;
 		little-endian;
 	};
 	opad0_regs: siul2_reg@0x4009D700 {
@@ -223,7 +223,7 @@
 	};
 	opad1_regs: siul2_reg@0x4403D70C {
 		compatible = "fsl,opad0_reqs-s32gen1","syscon","simple-mfd";
-		reg = <0x0 0x4403D70C 0x0 0x14>;
+		reg = <0x0 0x4403D70C 0x0 0x8>;
 		big-endian;
 	};
 	ipad1_regs: siul2_reg@0x4403D74C {
-- 
2.25.1

